python/ray/data/dataset.py

```diff
@@ -4825,7 +4825,7 @@ def deserialize_lineage(serialized_ds: bytes) -> "Dataset":
             .. testoutput::
 
                 Dataset(
-                   num_blocks=16,
+                   num_blocks=8,
```

Comment:
this number has to be equal to the core of the host machine, which is 8 now since we use medium size machine

Reply:
can we make this more robust? python can query the number of cores available programmatically, right?

Reply:
absolutely